src/routes/snippets.rs

```diff
@@ -77,6 +143,68 @@ pub fn create_snippet(
     create_snippet_impl(&**storage, &snippet, origin.path())
 }
 
+fn split_marker(mut snippets: Vec<Snippet>, limit: usize) -> (Option<String>, Vec<Snippet>) {
+    if snippets.len() > limit {
+        snippets.truncate(limit);
+        (snippets.last().map(|m| m.id.to_owned()), snippets)
+    } else {
+        (None, snippets)
+    }
+}
+
+#[allow(clippy::too_many_arguments)]
+#[get("/snippets?<title>&<syntax>&<tag>&<marker>&<limit>")]
+pub fn list_snippets<'o, 'h>(
+    storage: State<Box<dyn Storage>>,
+    origin: &'o Origin,
+    title: Option<String>,
+    syntax: Option<String>,
+    tag: Option<String>,
+    limit: Option<Result<PaginationLimit, ApiError>>,
```

Comment:
**n.b.** I needed to change that in order to propagate ApiError returned by `FromFormValue` trait to users.